src/Microsoft.AspNet.Hosting/Program.cs

```diff
@@ -23,31 +24,37 @@ public Program(IServiceProvider serviceProvider)
             _serviceProvider = serviceProvider;
         }
 
-        public void Main(string[] args)
+        public IHostingEngine CreateHost(IConfiguration config)
         {
-            var config = new Configuration();
-            if (File.Exists(HostingIniFile))
-            {
-                config.AddIniFile(HostingIniFile);
-            }
-            config.AddEnvironmentVariables();
-            config.AddCommandLine(args);
-
-            var engine = WebHost.CreateEngine(_serviceProvider, config);
+            var builder = new WebHostBuilder(_serviceProvider);
             var server = config.Get("server");
             if (server != null)
             {
-                engine.UseServer(server);
+                builder.UseServer(server);
             }
             var startup = config.Get("app");
```

Comment:
So we want config to win right?  Where config is now going to override anything set problematically before hand?  Or do we want config to be the defaults applied in the ctor?  and UseStartup will win?